File: game_server/ClientProtocol/Common/DinnerEnum/EDinnerJoinerType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace Common.DinnerEnum
{

/// <summary>
/// 赴宴对象类型
/// </summary>
public enum EDinnerJoinerType {
	NONE, //0 ==== 
	[InspectorName("PLAYER - idx[1] - 玩家")]
	PLAYER, //1 ==== 玩家
	[InspectorName("HERO - idx[2] - 大臣")]
	HERO, //2 ==== 大臣
}

public class EDinnerJoinerTypeComparer : IEqualityComparer<EDinnerJoinerType>{
	public bool Equals(EDinnerJoinerType x, EDinnerJoinerType y) { return x == y; }
	public int GetHashCode(EDinnerJoinerType obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 3;
}
}

File: game_server/ServerProtocol/src/Common/DinnerEnum/EDinnerJoinerType.java
package Common.DinnerEnum;

/*********
 * 赴宴对象类型
 **/
public enum EDinnerJoinerType {
	NONE, //0 ==== 
	PLAYER, //1 ==== 玩家
	HERO, //2 ==== 大臣
	;
public static final EDinnerJoinerType[]  EDinnerJoinerType_Values = EDinnerJoinerType.values();
public static final int EDinnerJoinerType_Length = EDinnerJoinerType_Values.length;
public static EDinnerJoinerType EDinnerJoinerType_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= EDinnerJoinerType_Length){ return null; }
	return EDinnerJoinerType_Values[_ivalue];
}
}

